java/arcs/android/sdk/host/ArcHostHelper.kt

```diff
@@ -129,7 +137,7 @@ class ArcHostHelper(
         block: suspend () -> Any?
     ) {
         val bundle = Bundle()
-        val result = block()
+        val result = try { block() } catch (e: Exception) { e }
```

Comment:
Catching all exceptions looks worrisome to me. This will catch even basic programmer mistakes, like null pointer exceptions (rare in kotlin, I know) or illegal argument exceptions. I'm not completely sure what code is run inside this block, but usually you want those kinds of errors to fail very loudly and bring down the component in question, rather than have them disappear into the ether.

Reply:
Would ArcHostException be sufficient?

Reply:
It's catching whatever startArc/stopArc/etc throw.  startArc() could fail for numerous issues, deep in the stack, do we want these to always blow up AiAiService? This is in essence, converting all exceptions into ArcHostException so that a client can determine what went wrong on a potentially remote host (imagine a multi-device scenario, an arc spanning chrome or remote cloud)

I could rethrow the exception after propagating it via the ResultReceiver so that the Service that embeds the ArcHost can still blow up?

(realistically, the Service won't be taken down, instead, onStartCommand() will throw an exception and it'll get swallowed up somewhere in logcat I think)

Reply:
Just to follow up, when you look up an Arc.status, it's just going to tell you Running, Stopped, Error, etc. When an ArcHost in another service, process, or remote throws an exception,  the goal was an additionally report an ArcHostException, so that you would get at least some information useful for debugging, without digging into a bunch of separate logs.  For AiAi, yes, it'll all be in adb logcat, but this ArcHostException is meant to be a core API for Allocator. 

So the real question is, where do you trap it. This is essentially an 'uncaught exception' handler reporter.  I would expect most ArcHost exceptions to be caught and turned into actual lifecycle methods (e.g. ParticleState.Failed).  However, Yuan had some failures of exceptions deep in the stack bubbling up. I still think it's useful from the point of view of the Allocator API to have a single place to catch and report these to the caller.

Reply:
Added a rethrow after sending the report back to the receiver.

Reply:
Thanks for explaining, that sounds reasonable. My main worries are that exceptions will go unnoticed, or that testing will be made difficult (e.g. if you're testing a failure condition, and the exception gets funneled off and quietly logged somewhere rather than thrown).

Reply:
No you're right that this should have re-thrown, that's an oversight. Good catch. My real intent was to make sure in the future, people starting Arcs over multi-devices had some unified way of discovering details about Errors that occurred on other devices.